app/models/user.rb

```diff
@@ -1,9 +1,33 @@
 class User < ApplicationRecord
+  attr_accessor :remember_token
   before_save{self.email = email.downcase}
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
   has_secure_password
-  validates :name, presence: true, length: {maximum: Settings.user.maximum_name}
-  validates :email, presence: true, length: {maximum: Settings.user.maximum_email},
+  validates :name, presence: true, length: {maximum: Setting.maximum_name}
+  validates :email, presence: true, length: {maximum: Setting.maximum_email},
   format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
-  validates :password, presence: true, length: {minimum: Settings.user.minimum_password}
+  validates :password, presence: true, length: {minimum: Setting.minimum_password}
+
+  def self.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def self.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def remember
+    self.remember_token = User.new_token
```

Comment:
chị bỏ self chỗ này dc ko

Reply:
theo e nghỉ thì bỏ được,
self là thằng user giôngs như this trong java.

Reply:
sao ko test thử, bỏ dc mới hay đó.
Nó mang ý nghĩa giống this trong java. Nhưng:
self ở đây là của object á. do `attr_accessor` khởi tạo ra nè, nó thuộc về object chứ ko thuộc về class...

tìm hiểu thêm về `attr_accessor` nhé (2tat)